packages/aws-cdk-lib/aws-ssm/lib/parameter.ts

```diff
@@ -472,6 +473,39 @@ export class StringParameter extends ParameterBase implements IStringParameter {
     return this.fromStringParameterAttributes(scope, id, { parameterName: stringParameterName });
   }
 
+  /**
+   * Imports an external string parameter by ARN.
+   */
+  public static fromStringParameterArn(scope: Construct, id: string, stringParameterArn: string): IStringParameter {
+    if (Token.isUnresolved(stringParameterArn)) {
+      throw new Error('stringParameterArn cannot be an unresolved token');
+    }
+
+    // has to be the same region
+    // split the arn string to get the region string
+    // arn sample: arn:aws:ssm:us-east-1:123456789012:parameter/dummyName
+    const arnParts = stringParameterArn.split(':');
+    const stackRegion = Stack.of(scope).region;
+    if (arnParts.length !== 6) {
+      throw new Error('unexpected StringParameterArn format');
+    } else if (!Token.isUnresolved(stackRegion) && arnParts[3] !== stackRegion) {
```

Comment:
can we enforce user to set a region in order to use this shared parameter or add something like a warning which states that same region is expected for this shared parameter arn, this is to ensure they don't see the failure directly at runtime.

Reply:
Yes we could. But if we do that, those who prefer stack environment-agnostic and is aware of this limitation would be forced to configure stack env vars. They would lose their freedom to be environment-agnostic.

Failing early is great. But if the cost is losing the freedom of environment-agnostic I would be hesitant to enforce that. wdyt?

Reply:
would adding a warning be forcing? i thought warning would just let them know what's about to happen in case they've not configured it and it will not fail synth or deploy, one can choose to ignore or take care of it.

Reply:
Sure. Just added an annotation warning when stack.region is an unresolved token. Let me know if it looks good to you.